File: awa/sdk_dist/python/awa/client/utils/activity/run_command_activity.py
from datetime import timedelta

from temporalio import workflow

from awa.client import constants
from awa.client.models.command_result import CommandInput, CommandResult


async def run_command_activity(command_input: CommandInput) -> CommandResult:
    """Run a command and return the result.

    Args:
        command_input: The command input containing command and working directory.

    Returns:
        CommandResult with success status and output.

    """
    result = await workflow.execute_activity(
        constants.ACTIVITY_RUN_COMMAND,
        arg=command_input,
        task_queue=constants.AWA_DEFAULT_TASK_QUEUE,
        start_to_close_timeout=timedelta(seconds=300),  # 5 minutes for commands
    )
    return CommandResult.model_validate(result)

File: awa/sdk_dist/csharp/Awa.Client/Utilities/Activity/RunCommandActivity.cs
using System;
using System.Threading.Tasks;
using Awa.Client.Constants;
using Awa.Client.Models;

namespace Awa.Client.Utilities.Activity
{
    public static partial class ActivityUtilities
    {
        /// <summary>
        /// Run a command and return the result.
        /// </summary>
        /// <param name="commandInput">The command input containing command and working directory.</param>
        /// <returns>CommandResult with success status and output.</returns>
        public static async Task<CommandResult?> RunCommandActivity(CommandInput commandInput)
        {
            var result = await Temporalio.Workflows.Workflow.ExecuteActivityAsync<CommandResult?>(
                AwaConstants.ActivityRunCommand,
                new object?[] { commandInput },
                new()
                {
                    TaskQueue = AwaConstants.AwaDefaultTaskQueue,
                    StartToCloseTimeout = TimeSpan.FromSeconds(300) // 5 minutes for commands
                });

            return result;
        }
    }
}
